man/systemd.exec.xml

```diff
@@ -846,6 +846,14 @@ CapabilityBoundingSet=~CAP_B CAP_C</programlisting>
         runtime, and are generally more expressive. For example, <varname>MemoryMax=</varname> is a more
         powerful (and working) replacement for <varname>LimitRSS=</varname>.</para>
 
+        <para>Note that <varname>LimitNPROC=</varname> will limit the number of processes from one (real) UID and
+        not the number of processes started (forked) by the service. Therefore the limit is cumulative for all
+        processes running under the same UID. Please also note that the <varname>LimitNPROC=</varname> will not be
+        enforced if the service is running as root (and not dropping privileges). To overcome these limitations and
+        actually limit the number of processes launched by a service, <varname>TasksMax=</varname> (see
+        <citerefentry><refentrytitle>systemd.resource-control</refentrytitle><manvolnum>5</manvolnum></citerefentry>)
+        should be used instead of <varname>LimitNPROC=</varname>.</para>
```

Comment:
I think adding this makes sense, but I'd tone this down a bit. i.e. not say "should be used", but say something like "is typically a better choice". 

Because we never know what people use LIMIT_NPROC for. For example, since it remains set across setuid() people might want to set it so that it still applies after a daemon changes uids later on.

hence, recommending people to use TasksMax= is good, but given we know so little abou thow people's usecase it should be a recommendation, not more

Reply:
Thank you for your feedback, I've changed the wording accordingly (and replaced the existing commit as recommended in `docs/CONTRIBUTING.md`).

Regarding daemons calling setuid: The `TasksMax=` limit will still apply after that, setuid will not take the process out of the cgroup.